### PR TITLE
refactor: rearrange flux-related storage code to match 2.x

### LINKED ISSUE #21112: Rearrange files under `storage/` to be consistent across 1.x and 2.x
The layout of flux-related code under `storage/` doesn't match between 1.x and 2.x, which makes back-/forward-porting fixes more difficult than it should be. Rearrange the files in `master-1.x` to match `master`. For example:
1. `storage/reads/flux_table.go` -> `storage/flux/table.go`
2. `storage/reads/flux_reader.go` -> `storage/flux/reader.go`

### PR DESCRIPTION
**Pull request by danxmoran**
Closes #21112

Builds off of  #21107 for now

Rearranges the flux-related files under `storage/` to match 2.x, so backports are less painful.

### CODE REVIEW

File: storage/flux/reader.go [hunk 1]
```diff
@@ -12,9 +12,11 @@ import (
 	"github.com/influxdata/flux/memory"
 	"github.com/influxdata/flux/plan"
 	"github.com/influxdata/flux/values"
-	"github.com/influxdata/influxdb/flux/stdlib/influxdata/influxdb"
+	query "github.com/influxdata/influxdb/flux/stdlib/influxdata/influxdb"
```
Comment (danxmoran) [inline storage/flux/reader.go:15]: I tried moving the relevant file (`storage.go`) out of `influxdb` and into `query`, but hit circular reference errors. Decided to leave it alone for now

File: storage/flux/types.tmpldata [hunk 2]
```diff
@@ -0,0 +1,32 @@
+[
```
Comment (lesam) [inline storage/flux/types.tmpldata:1]: forgot to delete the old file?

Comment (danxmoran) [reply storage/flux/types.tmpldata:1]: There are 2 copies in 2.x (ugh)

Comment (lesam) [reply storage/flux/types.tmpldata:1]: Let's fix that before merging...

**Review by lesam — APPROVED** [review]

**Pull request merged**